pkg/cloud/services/ec2/instances.go

```diff
@@ -294,8 +304,23 @@ func (s *Service) runInstance(role string, i *infrav1.Instance) (*infrav1.Instan
 		input.UserData = aws.String(base64.StdEncoding.EncodeToString(buf.Bytes()))
 	}
 
-	if len(i.SecurityGroupIDs) > 0 {
-		input.SecurityGroupIds = aws.StringSlice(i.SecurityGroupIDs)
+	if len(i.NetworkInterfaces) > 0 {
+		instances := make([]*ec2.InstanceNetworkInterfaceSpecification, len(i.NetworkInterfaces))
```

Comment:
It's not mandatory and the performance impact is probably negligible here, but it is indeed best practice.

Reply:
Yeah agreed, I was pointing out though that given that the line below uses `append`, we should set the length to `0`, otherwise we'll be appending at the end of the slice and have the first few with nil pointers